db/migrate/20210202141844_add_user_id_to_articles.rb

```diff
@@ -0,0 +1,5 @@
+class AddUserIdToArticles < ActiveRecord::Migration[6.1]
+  def change
+    add_column :articles, :user_id, :int
```

Comment:
this is a foreign key. So it is a good practice, tell this to the database.
see i.e. https://guides.rubyonrails.org/association_basics.html#the-belongs-to-association
this makes sure on a database level that your relations stay consistent. With a foreign key you also automatically add an index to this column. This is important for performance

Reply:
OK. Man braucht, wahrscheinlich dafür noch eine Migration.

Schreibt man dann: 
```
class MigrationName < ActiveRecord::Migration
  def change
    add_foreign_key :articles, :user
  end
end
```

[Source](https://stackoverflow.com/questions/16257116/adding-foreign-key-to-a-rails-model)

Reply:
Das wird schon stimmen. Ich weiß nicht. Ich muss bei sowas auch immer in der Doku nachlesen... :-)

Reply:
Oder einfach article.rb model so schreiben:
```
class Article < ApplicationRecord
  belongs_to :user, foreign_key: true
....
end
```

Reply:
nein, das geht so nicht.
An dieser Stelle wird ja kein Datenbankschema festgelegt. Das muss immer in einer Migration passieren.
Man kann an dieser Stelle sowas schreiben: `belongs_to: user, foreign_key: 'user_id' um festzulegen, welcher Fremdschlüssel genutzt werden soll, falls dies nicht eindeutig ist, falls es z.B. 2 Relationen zur gleichen Tabelle gibt. (z.B.author, reviewer, oder so). In der Regel ist dies aber nicht notwendig.

Reply:
z.B. :  `add_foreign_key :articles, :authors`
https://guides.rubyonrails.org/active_record_migrations.html#foreign-keys

Also in meinem Fall:
`add_foreign_key :articles :users`
Maybe?